### ISSUE TITLE
Slowness to generate proofs with recent zerokit versions

### ISSUE THREAD

**Issue by richard-ramos** [waku-org/js-rln#66]
As described in these comments:
- https://github.com/waku-org/js-rln/pull/58#issuecomment-1530121954
- https://github.com/waku-org/js-rln/pull/58#issuecomment-1532050914
- https://github.com/vacp2p/zerokit/pull/161#issuecomment-1584985557
The time to generate a proof increased a lot going from ~3s to ~23s, with this commit https://github.com/vacp2p/zerokit/commit/5eb98d4b330e1107ed18c22856707110302263e4 introducing the slowness

**Comment by weboko** [comment]
My understanding is that in this issue we should:
- [ ] investigate why it happens 
- [ ] mitigate on js side if possible

@richard-ramos is it correct?

**Comment by richard-ramos** [comment]
Indeed, but i'm not entirely sure we'll be able to mitigate the issue on JS side as it seems to have been introduced in zerokit

**Comment by rymnc** [comment]
@tyshko-rostyslav any idea why this is happening?

**Comment by tyshko-rostyslav** [comment]
If indeed this commit https://github.com/vacp2p/zerokit/commit/5eb98d4b330e1107ed18c22856707110302263e4 is the reason, it's quite odd, nothing special is happening there

**Comment by tyshko-rostyslav** [comment]
There was a significant slowdown, when we changed versions, specifically a lot of operations became compressed. As you've seen in this thread, I tried uncompressed versions in ark-circom 0.4, which somewhat helped, but not to the desired effect

**Comment by rymnc** [comment]
right, I think it'd be more performant if js-rln used a similar approach to https://github.com/Rate-Limiting-Nullifier/rlnjs, maybe even use the package to avoid de-duped work

**Comment by weboko** [comment]
I need to look into how we can leverage https://github.com/Rate-Limiting-Nullifier/rlnjs

**Comment by weboko** [comment]
`rlnjs` was not selected to be used due to fact that we needed to maintain proxy in JS code for our RLN contract and be able to optimize code in our own way.

Ice boxing this task as in future we might have a light weight way of generating / verifying proofs. Potentially we can use parts of `rlnjs` for this to avoid using `wasm` version of zerokit.

@adklempner @waku-org/js-waku-developers
